lib/resty/core/phase.lua

```diff
@@ -0,0 +1,57 @@
+local ffi = require 'ffi'
+local C = ffi.C
+local base = require "resty.core.base"
+local tostring = tostring
+
+ffi.cdef[[
+int ngx_http_lua_ffi_get_phase(ngx_http_request_t *r, char **err)
+]]
+
+
+local _M = {
+    _VERSION = base.version
+}
+
+
+local errmsg = base.get_errmsg_ptr()
+local context_lookup = {
```

Comment:
There is also this form we could use:

local context_lookup = {}
table.insert(context_lookup, 0x0001, "set")
table.insert(context_lookup, 0x0002, "rewrite")
table.insert(context_lookup, 0x0004, "access")
table.insert(context_lookup, 0x0008, "content")
table.insert(context_lookup, 0x0010, "log")
table.insert(context_lookup, 0x0020, "header_filter")
table.insert(context_lookup, 0x0040, "body_filter")
table.insert(context_lookup, 0x0080, "timer")
table.insert(context_lookup, 0x0100, "init_worker")
table.insert(context_lookup, 0x0200, "balancer")
table.insert(context_lookup, 0x0400, "ssl_cert")
table.insert(context_lookup, 0x0800, "ssl_session_store")
table.insert(context_lookup, 0x1000, "ssl_session_fetch")

This prevents us from needing to call `tostring()` below, I'm just not sure how we feel about arrays with empty indices. It also performs 3 times faster in the following micro benchmark:

```
require "resty.core"
for i = 1, 1000000 do
    phase = ngx.get_phase()
end
```

The form in the existing PR (string converted lookup) takes about 30000 microseconds to run; the form in this comment (integer lookup) takes about 10000 microseconds to run. Let me know what you think the best approach is here, if we need to be worried about this much micro performance analysis in this context, or if we should focus just on coding style/readability. Thanks!